### PR TITLE
Improve parser for --healthcheck-command

### PR DESCRIPTION
**Pull request by stefanb2**
* fix Docker CLI compatibility issue: the `--healthcheck-command` option
value should not be split but instead be passed as single string to
`/bin/sh -c <opt>`.

* on the other hand implement the same extension as is already available
for `--entrypoint`, i.e. allow the option value to be a JSON array of
strings. This will make life easier for tools like podman-compose.

Continuation of #3455 & #3507

### CODE REVIEW

**Comment by stefanb2** [comment]
@muayyad-alsadi FYI

File: docs/podman-run.1.md [hunk 1]
```diff
@@ -279,12 +279,14 @@ The example maps gids 0-2000 in the container to the gids 30000-31999 on the hos
...
 Add additional groups to run as
 
-**--healthcheck-command**=*command*
+**--healthcheck-command**=*"command"* | *'["command", "arg1", ...]'*
 
 Set or alter a healthcheck command for a container. The command is a command to be executed inside your
 container that determines your container health.  The command is required for other healthcheck options
 to be applied.  A value of `none` disables existing healthchecks.
 
+You need to specify multi option commands in the form of a json string.
```
Comment (mheon) [inline docs/podman-run.1.md]: Maybe reword to "Multiple options can be passed in the form of a JSON array; otherwise, the command will be interpreted as an argument to `/bin/sh -c`"

**Comment by mheon** [comment]
/approve

Code LGTM, manpages could use a little touchup

File: docs/podman-run.1.md (hunk 1 above)
Comment (stefanb2) [reply docs/podman-run.1.md]: Probably. I just copy&pasted the sentence from `--entrypoint` :-)

**Comment by stefanb2** [comment]
Instead of again hard-coding `/bin/sh -c` the latest patch set simply generates the 2-part list `"CMD-SHELL", <opt>`. This will be translated by `runHealthCheck()` to `/bin/sh -c "<opt>"`.

**Comment by rhatdan** [comment]
LGTM

**Comment by mheon** [comment]
Nice work, thanks!

/lgtm

**Comment by mheon** [comment]
Test failures look like infra flakes.

/retry

**Comment by mheon** [comment]
Oh, oops.
/retest

**Comment by mheon** [comment]
Prow seems like it's down.
